docs/core/howto/listings/servers/chat.py

```diff
@@ -24,16 +24,18 @@ def lineReceived(self, line):
 
     def handle_GETNAME(self, name):
         if name in self.users:
-            self.sendLine("Name taken, please choose another.")
+            self.sendLine(b"Name taken, please choose another.")
             return
-        self.sendLine(f"Welcome, {name}!")
+        self.sendLine(f"Welcome, {name.decode('utf-8')}!".encode("utf-8"))
         self.name = name
         self.users[name] = self
         self.state = "CHAT"
 
     def handle_CHAT(self, message):
-        message = f"<{self.name}> {message}"
-        for name, protocol in self.users.iteritems():
+        message = f"<{self.name.decode('utf-8')}> {message.decode('utf-8')}".encode(
+            "utf-8"
+        )
```

Comment:
Just a minor comment to avoid all the encode + decode

```suggestion
        message = b'<' + self.name  + '> ' + message
```

Reply:
I also agree with this comment but I think 
```python
message = b'<' + self.name  + b'> ' + message
``` 
is correct to avoid `TypeError`.
Is there any reason for that we didn't applied it?

Reply:
Hi. Thanks for the feedback. This was merged as it was better than what we had in `trunk`. There is alwasy place for improvement :)
Feel free to open a new GitHub Issue and PR if you would like to see this change merged.

The IRC implementation in Twisted needs some love and I would be happy to see people care about this code :)

Cheer

Reply:
Dear @adiroiban, I've created an issue and a pr for it.